lib/Doctrine/DBAL/Schema/Index.php

```diff
@@ -200,9 +200,7 @@ public function spansColumns(array $columnNames)
      */
     public function isFullfilledBy(Index $other)
     {
-        // allow the other index to be equally large only. It being larger is an option
-        // but it creates a problem with scenarios of the kind PRIMARY KEY(foo,bar) UNIQUE(foo)
```

Comment:
This is the bit that isn't clear.

A test is needed:

 * index `unique(foo)`
 * index `unique(foo, bar)`

The test should verify that the first unique index should not be dropped

Reply:
Thanks for explanation, another simple test case added.